ztp/ztp-policy-generator/kustomize/plugin/policyGenerator/v1/policygenerator/policyGen/policyBuilder.go

```diff
@@ -75,7 +69,7 @@ func (pbuilder *PolicyBuilder) Build(customResourseOnly bool) (map[string]interf
 				if resource["metadata"].(map[string]interface{})["namespace"] != nil {
 					name = name + "-" + resource["metadata"].(map[string]interface{})["namespace"].(string)
 				}
-				policies[ utils.CustomResource + "/" + name ] = resource
+				policies[ utils.CustomResource + "/" + pbuilder.PolicyGenTemp.Metadata.Name + "/" + name ] = resource
```

Comment:
Is there any chance that the Metadata name is empty or does not exist? If so, add some error handling for that case.

Reply:
it shouldn't be empty. You cannot create CR without name